### ISSUE TITLE
Revive does not type check by default (and I don't know how to set it)

### ISSUE THREAD

**Issue by damif94** [mgechev/revive#809]
**Describe the bug**
The typechecker, allegedly by default, is not actually enabled.

**To Reproduce**
1. I installed revive `go install github.com/mgechev/revive@latest`
2. I run it as follows:
```shell
# flags

REVIVE_FORCE_COLOR=1 revive -formatter friendly
```

without any config file

**Expected behavior**
The type checker should complaint about bad argument type

**Logs**

**Desktop (please complete the following information):**
 - OS: Macos monterrey M1
 - Version of Go 1.20

**Additional context**
<img width="933" alt="Screen Shot 2023-03-28 at 18 46 50" src="https://user-images.githubusercontent.com/29461526/228374157-540b66bf-bd91-42bc-9c71-d8b242afed4a.png">
No hint about type issues for the argument of `foo`

**Comment by chavacava** [comment]
Hola @damif94, GO being a static and strong typed language, the GO compiler will warn you on type mismatches. 
Therefore, linters, like `revive`, are not in charge of typechecking the source code.
 
You can configure your IDE  to use gopls (the implementation of the LSP for GO) and type problems will be highlighted without need of running the compiler.

**Comment by chavacava** [comment]
Clossing the issue. @damif94, do not hesitate to reopen it if my comment is not clear enough.
